src/main/java/tc/oc/pgm/freeze/FreezeManager.java

```diff
@@ -0,0 +1,124 @@
+package tc.oc.pgm.freeze;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.bukkit.ChatColor;
+import org.bukkit.Material;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemFlag;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import tc.oc.pgm.AllTranslations;
+import tc.oc.pgm.Config;
+import tc.oc.pgm.api.PGM;
+import tc.oc.pgm.api.Permissions;
+import tc.oc.pgm.api.chat.Sound;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.match.ObservingParty;
+
+/** Controls player freezing and those in freeze mode */
+public class FreezeManager {
+  private static final Sound FREEZE_SOUND = new Sound("mob.enderdragon.growl", 1f, 2f);
+
+  private static FreezeManager freezeManager;
+
+  public static FreezeManager get() {
+    if (freezeManager == null) freezeManager = new FreezeManager();
+    return freezeManager;
+  }
+
+  private Map<Player, Boolean> freezeModeMap = new HashMap<>();
+  private Set<Player> frozenPlayers = new HashSet<>();
+
+  public void assignFreezeItem(Player freezer) {
```

Comment:
There's some class called ObserverToolFactory or something, this should go there.